### PR TITLE
added crud for users/roles

### PR DESCRIPTION
**Pull request by dybrom**

### CODE REVIEW

File: src/main/java/com/example/fixba/config/SecurityConfig.java [hunk 1]
```diff
@@ -28,16 +31,19 @@
     @Override
     protected void configure(AuthenticationManagerBuilder auth) throws Exception {
         auth.userDetailsService(userDetailsService).passwordEncoder(bCryptPasswordEncoder);
-//        super.configure(auth);
```
Comment (vedadsurkovic) [inline src/main/java/com/example/fixba/config/SecurityConfig.java:31]: remove commented code

File: src/main/java/com/example/fixba/config/SecurityConfig.java [hunk 2]
```diff
@@ -28,16 +31,19 @@
...
 
     @Override
     protected void configure(HttpSecurity http) throws Exception {
+        CustomAuthenticationFilter customAuthenticationFilter = new CustomAuthenticationFilter(authenticationManagerBean());
+        customAuthenticationFilter.setFilterProcessesUrl("/api/login");
         http.csrf().disable();
         http.sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS);
-        http.authorizeRequests().antMatchers("/login").permitAll().anyRequest().permitAll();
-        http.addFilter(new CustomAuthenticationFilter(authenticationManagerBean()));
-//        super.configure(http);
```
Comment (vedadsurkovic) [inline src/main/java/com/example/fixba/config/SecurityConfig.java:40]: remove commented code

File: src/main/java/com/example/fixba/config/SecurityConfig.java [hunk 3]
```diff
@@ -28,16 +31,19 @@
     @Override
     protected void configure(AuthenticationManagerBuilder auth) throws Exception {
         auth.userDetailsService(userDetailsService).passwordEncoder(bCryptPasswordEncoder);
-//        super.configure(auth);
     }
 
     @Override
     protected void configure(HttpSecurity http) throws Exception {
+        CustomAuthenticationFilter customAuthenticationFilter = new CustomAuthenticationFilter(authenticationManagerBean());
```
Comment (vedadsurkovic) [inline src/main/java/com/example/fixba/config/SecurityConfig.java:42]: you can do this with one command, I can send you an example

File: src/main/java/com/example/fixba/config/SecurityConfig.java (hunk 2 above)
Comment (vedadsurkovic) [inline src/main/java/com/example/fixba/config/SecurityConfig.java:40]: remove commented code

File: src/main/java/com/example/fixba/filter/CustomAuthorizationFilter.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+import static java.util.Arrays.stream;
+import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
+
+@Slf4j
+public class CustomAuthorizationFilter extends OncePerRequestFilter {
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws ServletException, IOException {
+        filterChain.doFilter(request, response);
+//
```
Comment (vedadsurkovic) [inline src/main/java/com/example/fixba/filter/CustomAuthorizationFilter.java]: remove this

File: src/main/java/com/example/fixba/model/User.java [hunk 5]
```diff
@@ -1,21 +1,26 @@
...
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.List;
+import java.util.stream.Collectors;
 
 @Data
 @NoArgsConstructor
 @AllArgsConstructor
-@Entity(name= "users")
+@Entity(name = "users")
```
Comment (vedadsurkovic) [inline src/main/java/com/example/fixba/model/User.java]: move this name prop to @Table annotation

File: src/main/java/com/example/fixba/service/AuthService.java [hunk 6]
```diff
@@ -1,11 +1,8 @@
 package com.example.fixba.service;
 
-import com.example.fixba.dto.UserDTO;
 import com.example.fixba.dto.UserLoginDTO;
-import org.springframework.stereotype.Service;
 
-@Service
 public interface AuthService {
-    UserDTO register(UserLoginDTO userData);
+//    UserDTO register(UserLoginDTO userData);
```
Comment (vedadsurkovic) [inline src/main/java/com/example/fixba/service/AuthService.java]: remove this

File: src/main/java/com/example/fixba/service/AuthServiceImpl.java [hunk 7]
```diff
@@ -29,21 +28,19 @@ public AuthServiceImpl(UserRepository userRepository, PasswordEncoder passwordEn
...
-        List<Role> roles = new ArrayList<>();
-        for(Long id:userData.getRoles()) {
-            Role role = roleService.getRole(id);
-            roles.add(role);
-        }
-        User user = new User(userData.getEmail(), passwordEncoder.encode(userData.getPassword()), roles);
-        userRepository.save(user);
-        return DTOUtility.getUserDTOFromUser(user);
-    }
+//    public UserDTO register(UserLoginDTO userData) {
```
Comment (vedadsurkovic) [inline src/main/java/com/example/fixba/service/AuthServiceImpl.java]: remove this

File: src/main/java/com/example/fixba/util/DTOUtility.java [hunk 8]
```diff
@@ -1,18 +1,26 @@
...
-
 public class DTOUtility {
 
-    public static UserDTO getUserDTOFromUser(User user) {
-        return new UserDTO(user.getId(), user.getName(), user.getEmail(), user.getRoles());
-    }
-    public static UserDTO getNewUserDTOFromUser(User user) {
-        return new UserDTO(user.getId(), user.getEmail());
-    }
+//    public static UserDTO getUserDTOFromUser(User user) {
```
Comment (vedadsurkovic) [inline src/main/java/com/example/fixba/util/DTOUtility.java]: remove this

**Review by vedadsurkovic — APPROVED** [review]

File: src/main/java/com/example/fixba/config/SecurityConfig.java (hunk 1 above)
Comment (dybrom) [reply src/main/java/com/example/fixba/config/SecurityConfig.java:31]: this is showing that it is deleted

File: src/main/java/com/example/fixba/config/SecurityConfig.java (hunk 2 above)
Comment (dybrom) [reply src/main/java/com/example/fixba/config/SecurityConfig.java:40]: same here

File: src/main/java/com/example/fixba/controller/UserController.java [hunk 9]
```diff
@@ -1,42 +1,55 @@
...
+
+	UserService userService;
+
+	@Autowired
+	public UserController(UserService userService) {
+		this.userService = userService;
+	}
+
+	@Override
+	public ResponseEntity<List<UserContract>> getUsers(@Valid Integer page, @Valid Integer size, @Valid String searchTerm, @Pattern(regexp = "^(((dueDate|title|createdAt){1}:)((asc|desc){1}))(,{1}(((dueDate|title|createdAt)+:)((asc|desc){1})))*$") @Valid String order) {
```
Comment (AmerSurkovic) [inline src/main/java/com/example/fixba/controller/UserController.java:27]: Extract this regex to static string on top of the controller. Generally you should extract all string values on top of the class

File: src/main/java/com/example/fixba/service/RoleService.java [hunk 10]
```diff
@@ -1,14 +1,15 @@
...
 
+import com.example.fixba.generated.model.RoleAddContract;
+import com.example.fixba.generated.model.UserRoleContract;
 import com.example.fixba.model.Role;
-import org.springframework.stereotype.Service;
 
 import java.util.List;
 
-@Service
 public interface RoleService {
```
Comment (AmerSurkovic) [inline src/main/java/com/example/fixba/service/RoleService.java:9]: I would rename this to IRoleService and RoleServiceImpl to RoleService

Comment (AmerSurkovic) [reply src/main/java/com/example/fixba/service/RoleService.java:9]: Same for the other services

File: src/main/java/com/example/fixba/controller/UserController.java (hunk 9 above)
Comment (dybrom) [reply src/main/java/com/example/fixba/controller/UserController.java:27]: this here is generated code from open api scheme, how should i do this?

**Pull request merged**